### PR TITLE
event bus fire and don't care

### PR DESCRIPTION
**Pull request by sergigp**

### CODE REVIEW

File: src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala [hunk 1]
```diff
@@ -3,37 +3,33 @@ package com.sergigp.quasar.event
...
+  override def publish[E <: Event](event: E): Future[Unit] = {
+    val results = handlers.getOrElse(event.getClass, List.empty[Event => Future[Unit]]).map { handler =>
+      handleEvent(event, handler)
+    }
+    Future.sequence(results).map(_ => ()).recoverWith {
+      case _ => Future.successful(())
     }
+  }
 
   override def subscribe[E <: Event: ClassTag](handler: E => Future[Unit]): Unit = {
```
Comment (rgomezcasas) [inline src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:23]: Could this be injected instead of mutating every time?

**Review by jorgeavilacardenosa — APPROVED** [review]

File: src/main/scala/com/sergigp/quasar/event/EventBus.scala [hunk 2]
```diff
@@ -3,5 +3,5 @@ package com.sergigp.quasar.event
 import scala.reflect.ClassTag
 
 trait EventBus[P[_]] extends EventPublisher[P] {
-  def subscribe[C <: Event: ClassTag](handler: C => P[Unit]): Unit
+  def subscribe[E <: Event: ClassTag](handler: E => P[Unit]): Unit
```
Comment (rgomezcasas) [inline src/main/scala/com/sergigp/quasar/event/EventBus.scala:6]: 👏

**Review by rgomezcasas — COMMENTED** [review]
Some comments :)

File: src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala (hunk 1 above)
Comment (sergigp) [reply src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:23]: what will happen if a handler needs the eventbus as a dependency? :)

Comment (rgomezcasas) [reply src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:23]: If it's a singleton, nothing 😬✌️

Comment (rgomezcasas) [reply src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:23]: (or lazy 💤)

**Pull request merged**